src/persist.ts

```diff
@@ -34,17 +35,25 @@ const createPersistNode = async (
   persistEntry: PersistEntry
 ) => {
   const [node, configProp] = persistEntry;
-  const config: PersistConfig = Object.assign({ duration: 0 }, configProp);
+  const config: PersistConfig = { delay: 0, ...configProp };
 
   const persistStoreNode = PersistStoreNode.create({
     key: config.key,
   });
   persistStore.addNode(persistStoreNode);
 
   let timeout: ReturnType<typeof setTimeout> | number;
-  onSnapshot(node, (snapshot) => {
+  onSnapshot(node, (snapshotArg) => {
     clearTimeout(timeout);
+
+    const snapshot = { ...snapshotArg };
     timeout = setTimeout(() => {
+      for (const elem in snapshot) {
+        if (config.whitelist && config.whitelist.indexOf(elem) === -1) {
+          delete snapshot[elem];
+        }
+      }
```

Comment:
If `!config.whitelist`, this is pure overhead. Probably worth nesting this, maybe something like:

```ts
if (config.whitelist) {
  for (const elem in snapshot) {
    if (config.whitelist.indexOf(elem) === -1) delete snapshot[elem]
  }
}
```